src/globalvars.cpp

```diff
@@ -232,23 +265,40 @@ std::unordered_set<std::string> get_section_names()
   return ret;
 }
 
-void GlobalVars::add_known_global_var(const std::string_view &name)
+void GlobalVars::add_known(const std::string_view &name)
 {
   if (!GLOBAL_VAR_CONFIGS.contains(name)) {
     LOG(BUG) << "Unknown global variable: " << name;
   }
   auto str_name = std::string(name);
 
-  if (global_var_map_.contains(str_name)) {
+  if (added_global_vars_.contains(str_name)) {
     return;
   }
-  global_var_map_[std::move(str_name)] = GLOBAL_VAR_CONFIGS.at(name);
+  added_global_vars_[std::move(str_name)] = GLOBAL_VAR_CONFIGS.at(name);
+}
+
+void GlobalVars::add_named_param(const std::string &name,
+                                 Type type,
+                                 bool is_bool,
+                                 const std::string &default_val)
+{
+  if (added_global_vars_.contains(name)) {
+    return;
+  }
+  added_global_vars_[name] = GlobalVarConfig({
```

Comment:
Per comment elsewhere, this is actually much cleaner if you just pack the variant and unpack it here. There's a lot of extra information being carried here that isn't supported, and a lot of invalid states (e.g. is_bool = true & type != Type::integer) that indicates you're not using types sensible. As an alternative to the variant, you could wrap the variant in a `NamedParamType` simple class without a `global_var_type()` method that returns this too.